app/controllers/items/detail.js

```diff
@@ -337,6 +336,84 @@ export default GoodcityController.extend(
       }
     },
 
+    async deleteAndAssignNew(packageType) {
+      const item = this.get("item");
+      const type = item.get("detailType");
+      const detailId = item.get("detailId");
+      if (type) {
+        await this.runTask(
+          this.get("subformDetailService").deleteDetailType(type, detailId)
+        );
+      }
+      return this.assignNew(packageType, {
+        deleteDetailId: !this.isSubformPackage(packageType)
+      });
+    },
+
+    warnAndAssignNew(pkgType) {
+      const existingPkgType = this.get("item.code");
+      const packageName = existingPkgType.get("name");
+      const newPackageName = pkgType.get("name");
+      const translation = this.get("i18n");
+
+      this.get("messageBox").custom(
+        translation.t("items.new.subform.delete_subform_waring", {
+          newPackageName,
+          packageName
+        }),
+        translation.t("not_now"),
+        null,
+        translation.t("continue"),
+        () => {
+          this.deleteAndAssignNew(pkgType);
+        }
+      );
+    },
```

Comment:
This method is really verbose and not very useful. You could perhaps add a helper for that (maybe in the async service??).

e.g

```javascript
this.withConfirmation('"items.new.subform.delete_subform_waring").runTask(() => {
  return this.deleteAndAssignNew(pkgType);
});
```